pantheon-bundle/src/main/java/com/redhat/pantheon/extension/Events.java

```diff
@@ -33,21 +31,15 @@ public Events(@Reference JobManager jobManager) {
         this.jobManager = jobManager;
     }
 
-    /**
-     * Fires an event indicating that a module version has been publsihed. As its name implies,
-     * this event should be fired only after a module is published.
-     * @param moduleVersion The module version which has just been published.
-     */
-    public void fireModuleVersionPublishedEvent(ModuleVersion moduleVersion) {
-        ModuleVersionPublishedEvent event = new ModuleVersionPublishedEvent(moduleVersion.getPath());
+    public void fireEvent(Event evt) {
         Map<String, Object> props = newHashMap();
-        props.put(Event.class.getName(), event);
-        Job job = jobManager.createJob(MODULE_POST_PUBLISH_EVENT)
+        props.put(Event.class.getName(), evt);
+        Job job = jobManager.createJob(EVENT_TOPIC_NAME)
                 .properties(props)
                 .add();
 
         if(job == null) {
-            throw new RuntimeException("Something went wrong creating a " + MODULE_POST_PUBLISH_EVENT + " job");
+            throw new RuntimeException("Something went wrong firing a " + evt.getClass().getName() + " event");
```

Comment:
What do you suggest? This is happening at the point where the event is being fired, and a failure here will probably mean some fundamental piece of infrastructure is missing or misconfigured. This method is synchronous which also means that whatever operation is invoking it will also fail (expected).

I guess we could create a queue to handle retries, but that wouldn't guarantee that the event would be successfully fired eventually.